pyk/src/pyk/__main__.py

```diff
@@ -427,5 +432,50 @@ def exec_parse_outer(options: ParseOuterOptions) -> None:
         sys.stdout.write(f'{result_text}\n')
 
 
+def exec_kompilex(options: KompileXCommandOptions) -> None:
```

Comment:
Are there self-contained logical units in this function that could be extracted into helpers?

Reply:
There are similar operations in `exec_parse_outer` for resolving include paths and the main module name. I could see if they could be made into some utilities.

Reply:
I made a refactor to `KDefinitionOptions` to help reduce some of the duplicated logic. Let me know if it looks good to you.